tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -1547,6 +1547,43 @@ static LogicalResult Verify(DynamicSliceOp op) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// RealDynamicSliceOp
+//===----------------------------------------------------------------------===//
+// Verifies that operand rank matches start_indices/limit_indices/strides size
+static LogicalResult Verify(RealDynamicSliceOp op) {
+  auto input_type = op.operand().getType().dyn_cast<RankedTensorType>();
+  // If operand is unranked, there is very little to verify statically.
+  if (!input_type) {
+    return success();
+  }
+  int input_rank = input_type.getRank();
+
+  auto start_type = op.start_indices().getType().dyn_cast<RankedTensorType>();
+  auto limit_type = op.limit_indices().getType().dyn_cast<RankedTensorType>();
+  auto strides_type = op.strides().getType().dyn_cast<RankedTensorType>();
```

Comment:
Can you replace the dyn_cast with cast here? It is an anti-pattern to use dyn_cast and not check.

Reply:
done